src/main/java/com/mcserby/playground/javademoplayground/service/Dex.java

```diff
@@ -95,8 +95,8 @@ public ExchangeResult swap(ExchangeRequest request) {
             double k = x * y;
 
             double delta = exchangedValue;
-            double returnedValue = 0;
-            double exchangePoolNewPrice = 0;
+            double returnedValue;
+            double exchangePoolNewPrice;
```

Comment:
This logic can be refactored to use pure functions, limiting side effects. Please check lambda calculus paper of Alonzo Church https://en.wikipedia.org/wiki/Lambda_calculus form 1930s and stop writing imperative code like we live in the 50'.